SCClassLibrary/Common/Math/Integer.sc

```diff
@@ -171,6 +171,14 @@ Integer : SimpleNumber {
 		*/
 	}
 
+	factorial {
+		if(this > 12) {
+			Error("Insufficient integer precision for this number (factorial(%)). "
+				"Try to use a float instead.".format(this)).throw
```

Comment:
typical error message notation is `[component:] function: problem`. So a more consistent wording would be "factorial: insufficient integer precision for this number (%). Use Float.factorial instead." I'll add a note to the developer guide